### PR TITLE
fix: improve test speed

### PR DESCRIPTION
**Pull request by gengjiawen**

### CODE REVIEW

File: scripts/test-react-native-setup.sh [hunk 1]
```diff
@@ -13,7 +15,7 @@ buck build ReactAndroid/src/main/java/com/facebook/react/shell
 
 echo "Build React Native via Gradle"
 yarn install
-./gradlew --no-daemon :ReactAndroid:packageReactNdkLibsForBuck
+./gradlew :ReactAndroid:packageReactNdkLibsForBuck
 
 echo "Assemble RNTester app"
-./gradlew --no-daemon :packages:rn-tester:android:app:assembleRelease
```
Comment (cortinico) [inline scripts/test-react-native-setup.sh:19]: I would keep the `--no-daemon` flag here. Any reason for removing it?

Comment (gengjiawen) [reply scripts/test-react-native-setup.sh:19]: For build speed, but looks github action still timeout with.

Comment (cortinico) [reply scripts/test-react-native-setup.sh:19]: IMHO the build speed we gain out of the daemon is not worth the loss in isolation

**Review by cortinico — APPROVED** [review]

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -9,10 +9,11 @@ on:
   pull_request:
     paths: 
       - Dockerfile
+      - scripts/**
 
 jobs:
     build:
-      runs-on: ubuntu-20.04
+      runs-on: macos-11
```
Comment (cortinico) [inline .github/workflows/build.yaml:16]: Not sure we want to run this on Mac. What's the reason behind this?

Comment (gengjiawen) [reply .github/workflows/build.yaml]: Because linux is out of space when running test. But looks macOS not having docker installed.

Comment (gengjiawen) [reply .github/workflows/build.yaml]:
```
Run df -h
Filesystem      Size  Used Avail Use% Mounted on
/dev/root        84G   53G   32G  [6](https://github.com/react-native-community/docker-android/runs/6121950579?check_suite_focus=true#step:3:6)3% /
devtmpfs        3.4G     0  3.4G   0% /dev
tmpfs           3.4G  4.0K  3.4G   1% /dev/shm
tmpfs           695M  1.1M  694M   1% /run
tmpfs           5.0M     0  5.0M   0% /run/lock
tmpfs           3.4G     0  3.4G   0% /sys/fs/cgroup
/dev/loop0       62M   62M     0 100% /snap/core20/1405
/dev/loop1       68M   68M     0 100% /snap/lxd/22[7](https://github.com/react-native-community/docker-android/runs/6121950579?check_suite_focus=true#step:3:7)53
/dev/loop2       44M   44M     0 100% /snap/snapd/15177
/dev/sda15      105M  5.2M  100M   5% /boot/efi
/dev/sdb1        14G  4.1G  [9](https://github.com/react-native-community/docker-android/runs/6121950579?check_suite_focus=true#step:3:9).0G  32% /mnt
```

File: scripts/test-react-native-setup.sh [hunk 3]
```diff
@@ -13,7 +15,11 @@ buck build ReactAndroid/src/main/java/com/facebook/react/shell
...
 
-echo "Assemble RNTester app"
-./gradlew --no-daemon :packages:rn-tester:android:app:assembleRelease
+./gradlew :ReactAndroid:assembleDebug
+
+# ./gradlew --no-daemon :ReactAndroid:packageReactNdkLibsForBuck
+
+# echo "Assemble RNTester app"
+# ./gradlew --no-daemon downloadAll
+# ./gradlew --no-daemon :packages:rn-tester:android:app:assembleRelease
```
Comment (cortinico) [inline scripts/test-react-native-setup.sh:25]: Please note that those are still needed

Comment (gengjiawen) [reply scripts/test-react-native-setup.sh:25]: Yeap, but run all those will timeout on github action.
You can see https://github.com/react-native-community/docker-android/actions/runs/2745390327. 6 hours still not finished.

Comment (cortinico) [reply scripts/test-react-native-setup.sh]: Mmm that's really odd. I can't see on which Gradle task it got stuck actually

Comment (cortinico) [reply scripts/test-react-native-setup.sh]: For Android not. Also Hermes needs to be built at a certain point and the Docker image should contain all the necessary tooling to build it so I won't try to use prebuilts here

Comment (gengjiawen) [reply scripts/test-react-native-setup.sh]: > For Android not. Also Hermes needs to be built at a certain point and the Docker image should contain all the necessary tooling to build it so I won't try to use prebuilts here

If we build from source, the github action will just timeout. Not sure there is a better way (Ninja on android or limit cpu arch on build arm64 maybe)?
cc @Huxpro

Comment (cortinico) [reply scripts/test-react-native-setup.sh]: > Not sure there is a better way 

The easiest would be to build per arch and split the job in 4

**Pull request closed without merging**